File: app/src/main/java/com/blog/demo/control/list/ListViewCustomAdapterActivity.java
package com.blog.demo.control.list;

import android.app.Activity;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.widget.ListView;

import com.blog.demo.People;
import com.blog.demo.R;

import java.util.Arrays;

public class ListViewCustomAdapterActivity extends Activity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_control_list_view);

        ListView listView = findViewById(R.id.list_view);
        listView.setAdapter(new CustomAdapter(this, Arrays.asList(new People[]{
                new People("Peter", "ShangHai"),
                new People("Lily", "BeiJing"),
                new People("Jack", "GuangZhou"),
                new People("Mike", "ShengZhen")})));
    }

}

File: app/src/main/java/com/blog/demo/control/list/ListViewCustomAdapterActivity.kt
package com.blog.demo.control.list

import android.app.Activity
import android.os.Bundle
import android.widget.ListView
import com.blog.demo.People
import com.blog.demo.R

class ListViewCustomAdapterActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_control_list_view)
        val listView: ListView = findViewById(R.id.list_view)
        listView.adapter = CustomAdapter(this,
                arrayOf(
                    People("Peter", "ShangHai"),
                    People("Lily", "BeiJing"),
                    People("Jack", "GuangZhou"),
                    People("Mike", "ShengZhen")
                ).asList()
            )
    }

}